ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuationRuleTest.kt

```diff
@@ -0,0 +1,811 @@
+                val foo1 = requireNotNull(
+                    bar.filter { it == "bar" }
+                ) { "some message" }
+                val foo2 = requireNotNull(
+                    bar?.filter {
+                        it == "bar"
+                    }
+                ) { "some message" }
+                """.trimIndent()
+            chainMethodContinuationRuleAssertThat(code)
+                .setMaxLineLength()
+                .addAdditionalRuleProvider { ArgumentListWrappingRule() }
+                .addAdditionalRuleProvider { FunctionLiteralRule() }
+                .hasLintViolations(
+                    // Lint violation below will not be triggered during format as argument-list-wrapping rule prevents this error from occuring
+                    LintViolation(2, 30, "Expected newline before '.'"),
+                    LintViolation(3, 30, "Expected newline before '?.'"),
+                ).isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given the argument-list-wrapping, function-literal, multiline-expression and indent rules are enabled`() {
+            val code =
+                """
+                // $MAX_LINE_LENGTH_MARKER   $EOL_CHAR
+                val foo1 = requireNotNull(bar.filter { it == "bar" }) { "some message" }
+                val foo2 = requireNotNull(bar?.filter { it == "bar" }) { "some message" }
+                """.trimIndent()
+            val formattedCode =
+                """
+                // $MAX_LINE_LENGTH_MARKER   $EOL_CHAR
+                val foo1 =
+                    requireNotNull(
+                        bar.filter {
+                            it == "bar"
+                        }
+                    ) { "some message" }
+                val foo2 =
+                    requireNotNull(
+                        bar?.filter {
+                            it == "bar"
+                        }
+                    ) { "some message" }
+                """.trimIndent()
+            chainMethodContinuationRuleAssertThat(code)
+                .setMaxLineLength()
+                .addAdditionalRuleProvider { ArgumentListWrappingRule() }
+                .addAdditionalRuleProvider { FunctionLiteralRule() }
+                .addAdditionalRuleProvider { MultilineExpressionWrapping() }
+                .addAdditionalRuleProvider { IndentationRule() }
+                .isFormattedAs(formattedCode)
+                .hasLintViolations(
+                    // Lint violation below will not be triggered during format as argument-list-wrapping rule prevents this error from occuring
+                    LintViolation(2, 30, "Expected newline before '.'"),
+                    LintViolation(3, 30, "Expected newline before '?.'"),
+                ).isFormattedAs(formattedCode)
+        }
+    }
+
+    @Test
+    fun `Given a single line chained expression, not exceeding max line length, with too many chain operators`() {
```

Comment:
we can also add TC for binary operators on two long chains exceeding the max line lenght

Reply:
Those are ignore as explained in the rule:
```
    private fun ChainedExpression.exceedsMaxLineLength() =
        with(rootASTNode) {
            if (treeParent.elementType == BINARY_EXPRESSION) {
                // Chained expressions which are enclosed inside a binary expression are skipped for now. It depends on the situation
                // whether wrapping on the binary expression takes precedence on the chained expression or vice versa.
                // This can be illustrated with following examples:
                //    - wrapping on chained expression first makes more sense in this case
                //        val foo = listOf("one", "two", "three").filter { it.length > 3 }
                //      resulting in:
                //        val foo = listOf("one", "two", "three")
                //            .filter { it.length > 3 }
                //    - wrapping on binary expression first makes more sense in this case
                //        if (someOtherExpression || listOf("one", "two", "three").any { it.length > 3 }) { ... }
                //      resulting in:
                //        if (someOtherExpression ||
                //            listOf("one", "two", "three").any { it.length > 3 }
                //        ) { ... }
                false
            } else {
```